core/src/impl/Kokkos_FunctorAnalysis.hpp

```diff
@@ -126,7 +127,7 @@ struct FunctorAnalysis {
 
   template <typename F, typename = std::false_type>
   struct has_value_type {
-    using type = void;
+    using type = OverrideValueType;
```

Comment:
This is the most important change. Passing in a non-void `OverrideValueType` skips the value type deduction.

Reply:
Done.